src/Symfony/Component/Translation/Tests/Exception/UnsupportedSchemeExceptionTest.php

```diff
@@ -0,0 +1,86 @@
+<?php
+
+/*
+ * This file is part of the Symfony package.
+ *
+ * (c) Fabien Potencier <fabien@symfony.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Symfony\Component\Translation\Tests\Exception;
+
+use PHPUnit\Framework\TestCase;
+use Symfony\Bridge\PhpUnit\ClassExistsMock;
+use Symfony\Component\Translation\Bridge\Loco\Provider\LocoProviderFactory;
+use Symfony\Component\Translation\Exception\UnsupportedSchemeException;
+use Symfony\Component\Translation\Provider\Dsn;
+
+final class UnsupportedSchemeExceptionTest extends TestCase
+{
+    public static function setUpBeforeClass(): void
+    {
+        ClassExistsMock::register(__CLASS__);
+    }
+
+    protected function setUp(): void
+    {
+        ClassExistsMock::withMockedClasses([
+            LocoProviderFactory::class => false,
+        ]);
+    }
+
+    /**
+     * @dataProvider messageWhereSchemeIsPartOfSchemeToPackageMapProvider
+     */
+    public function testMessageWhereSchemeIsPartOfSchemeToPackageMap(string $scheme, string $package)
+    {
+        $dsn = new Dsn(sprintf('%s://localhost', $scheme));
+
+        $this->assertSame(
+            sprintf('Unable to synchronize translations via "%s" as the provider is not installed; try running "composer require %s".', $scheme, $package),
+            (new UnsupportedSchemeException($dsn))->getMessage()
+        );
+    }
+
+    public function messageWhereSchemeIsPartOfSchemeToPackageMapProvider(): \Generator
+    {
+        yield ['loco', 'symfony/loco-translation-provider'];
+    }
+
+    /**
+     * @dataProvider messageWhereSchemeIsNotPartOfSchemeToPackageMapProvider
+     */
+    public function testMessageWhereSchemeIsNotPartOfSchemeToPackageMap(string $expected, Dsn $dsn, string $name, array $supported)
+    {
+        $this->assertSame(
+            $expected,
+            (new UnsupportedSchemeException($dsn, $name, $supported))->getMessage()
+        );
+    }
+
+    public function messageWhereSchemeIsNotPartOfSchemeToPackageMapProvider(): \Generator
+    {
+        yield [
+            'The "somethingElse" scheme is not supported.',
+            new Dsn('somethingElse://localhost'),
+            null,
+            [],
+        ];
+
+        yield [
+            'The "somethingElse" scheme is not supported.',
+            new Dsn('somethingElse://localhost'),
+            'foo',
```

Comment:
This `$name` as no effect on the output. Is that correct?

Reply:
Yes, only if `$name` and `$supported` is set 👍